terminology/src/main/java/au/csiro/pathling/terminology/caching/InMemoryCachingTerminologyService.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 Commonwealth Scientific and Industrial Research
+ * Organisation (CSIRO) ABN 41 687 119 230.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package au.csiro.pathling.terminology.caching;
+
+import au.csiro.pathling.config.HttpClientCachingConfiguration;
+import au.csiro.pathling.fhir.TerminologyClient;
+import java.io.Closeable;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.infinispan.Cache;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+/**
+ * A terminology service that uses embedded Infinispan to cache results in-memory.
+ *
+ * @author John Grimes
+ */
+public class InMemoryCachingTerminologyService extends CachingTerminologyService {
+
+  public InMemoryCachingTerminologyService(@Nonnull final TerminologyClient terminologyClient,
+      @Nullable final Closeable toClose,
+      @Nonnull final HttpClientCachingConfiguration configuration) {
+    super(terminologyClient, toClose, configuration);
+  }
+
+  @Override
+  protected EmbeddedCacheManager buildCacheManager() {
+    return new DefaultCacheManager();
+  }
+
+  @Override
+  protected Cache<Integer, ?> buildCache(@Nonnull final EmbeddedCacheManager cacheManager,
+      @Nonnull final String cacheName) {
+    cacheManager.defineConfiguration(cacheName, new ConfigurationBuilder().build());
+    return cacheManager.getCache(cacheName);
+  }
```

Comment:
No, this was just a mistake - I have passed the configuration in correctly in 4e264f095493b3f96208cafdcd4fdd46a36b110f.

Reply:
Actually, it wasn't quite that simple 🙂

For Infinispan, these parameters only really control the amount of memory used by the cache. Maximum size can only be used when using persistence, and concerns the total size of the cache. Maximum entries is used when using in memory caching.

Eviction means that the cached data is moved out of memory. Persisted data doesn't get removed until cache expiry. We don't expire our cache entries because they contain the information that we need to revalidate them. I guess that means that the disk space could be unbounded.

I don't have any solution to this at this point. But having seen the behaviour of the solution so far, the persisted cache data takes up very little space on disk now that we have moved away from HTTP-level caching.

Reply:
Actually, I've had to remove the max size configuration for now as it requires additional configuration relating to binary encoding. I don't think this straightforward or desirable.